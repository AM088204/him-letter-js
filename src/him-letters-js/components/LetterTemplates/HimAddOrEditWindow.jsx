import React from 'react';
import CollapsibleMenuView from 'terra-collapsible-menu-view';
import { DisclosureManagerHeaderAdapter } from 'terra-disclosure-manager';
import Tabs from 'terra-tabs';
import  '../../stylesheets/HimLetterTemplates.css';

const HimAddOrEditWindow = (props) => {
  const { name } = props;
  const PropertiesTab = (
    <Tabs.Pane label="Properties" key="Properties" />
  );
  const PrintTemplateTab = (
    <Tabs.Pane label="Print Template" key="Print Template" />
   );
   const EmailTemplateTab = (
   <Tabs.Pane label="Email Template" key="Email Template" />
   );
  return (
    <div>
      <DisclosureManagerHeaderAdapter
        title={name}
        collapsibleMenuView={(
          <CollapsibleMenuView />
        )}
      />
      <div className='margin-top'>
        <Tabs>
          {PropertiesTab}
          {PrintTemplateTab}
          {EmailTemplateTab}
        </Tabs>
      </div>
    </div>
  );

}
export default HimAddOrEditWindow;