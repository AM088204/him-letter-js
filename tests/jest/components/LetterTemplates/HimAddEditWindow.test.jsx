import React from 'react';
import { shallowWithIntl, mountWithIntl } from 'terra-enzyme-intl';
import HimAddOrEditWindow from '../../../../src/him-letters-js/components/LetterTemplates/HimAddOrEditWindow';
import Tabs from 'terra-tabs';
import { DisclosureManagerHeaderAdapter } from 'terra-disclosure-manager';

describe('himAddOrEditWindow', () => {

    it('Should Render himAddOrEditWindow Button', () => {
        const himAddOrEditWindow = shallowWithIntl(<HimAddOrEditWindow />);
        expect(himAddOrEditWindow.first().shallow()).toMatchSnapshot();
    });

   
    it('Should Render himAdd', () => {
        const himAddOrEditWindow = shallowWithIntl(<HimAddOrEditWindow />);
        console.debug(himAddOrEditWindow.find(Tabs.Pane).at(0).props().label);
    });


    it('Should Render himAdd', () => {
        const himAddOrEditWindow = mountWithIntl(<HimAddOrEditWindow />);
        expect(himAddOrEditWindow.find(DisclosureManagerHeaderAdapter).props().collapsibleMenuView.type.displayName).toMatch('CollapsibleMenuView');
    });
});