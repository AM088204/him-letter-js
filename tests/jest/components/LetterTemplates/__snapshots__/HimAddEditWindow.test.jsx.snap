// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`himAddOrEditWindow Should Render himAddOrEditWindow Button 1`] = `
<div>
  <DisclosureManagerHeaderAdapter
    collapsibleMenuView={<InjectIntl(CollapsibleMenuView) />}
  />
  <div
    className="margin-top"
  >
    <Tabs
      fill={false}
      responsiveTo="parent"
      tabFill={false}
    >
      <TabPane
        isActive={false}
        isDisabled={false}
        isIconOnly={false}
        key="Properties"
        label="Properties"
      />
      <TabPane
        isActive={false}
        isDisabled={false}
        isIconOnly={false}
        key="Print Template"
        label="Print Template"
      />
      <TabPane
        isActive={false}
        isDisabled={false}
        isIconOnly={false}
        key="Email Template"
        label="Email Template"
      />
    </Tabs>
  </div>
</div>
`;
